### PR TITLE
Color display

### PR DESCRIPTION
**Pull request by shannonmoranetz**
- Added random color functionality 
- Added load palette & project functionality

### CODE REVIEW

File: src/components/PaletteControls/PaletteControls.js [hunk 1]
```diff
@@ -1,14 +1,58 @@
...
 export class PaletteControls extends Component {
+
+  findPaletteName = () => {
+    const  { palettes, currentPaletteId } = this.props;
+    const matchingPalette = palettes.find((palette) => {
+      return palette.id === currentPaletteId
+    })
+    return matchingPalette.name;
+  }
+
```
Comment (hlhartley) [inline src/components/PaletteControls/PaletteControls.js:14]: Can we go over this please?

File: src/components/PaletteControls/PaletteControls.js [hunk 2]
```diff
@@ -1,14 +1,58 @@
...
-        <button className="generate-button">Generate New Palette!</button>
-        <h2 className="palette-header">PALETTE 1</h2>
+        {this.props.palettes.length ? (
+          <div className="palette-controls-container">
+            <button className="generate-button" onClick={this.generateRandomColors}>Generate New Palette!</button>
+            <h2 className="palette-header">{this.findPaletteName()}</h2>
+          </div>
+        ) : (
+          <p>loading...</p>
+        )}
```
Comment (hlhartley) [inline src/components/PaletteControls/PaletteControls.js:41]: Maybe we could add a loading icon in here

File: src/components/PaletteControls/PaletteControls.js [hunk 3]
```diff
@@ -1,14 +1,58 @@
...
+  palettes: state.palettes,
+  currentPaletteId: state.currentPaletteId,
+  randomHexcodes: state.randomHexcodes
+});
+
+export const mapDispatchToProps = (dispatch) => ({
+  setRandomHexcodes: (hexcodes) => dispatch(setRandomHexcodes(hexcodes)),
+  toggleDisplayRandom: (shouldDisplay) => dispatch(toggleDisplayRandom(shouldDisplay))
+});
+
```
Comment (hlhartley) [inline src/components/PaletteControls/PaletteControls.js:57]: We could add propTypes at the bottom

File: src/components/ProjectCard/ProjectCard.js [hunk 4]
```diff
@@ -16,31 +18,64 @@ export class ProjectCard extends Component {
...
         return (
           <div key={uuid()} className="mapped-palettes">
-            <p className="palette-name">{name}</p>
-            <p className="color-minibox">color1: {color1}</p>
-            <p className="color-minibox">color2: {color2}</p>
-            <p className="color-minibox">color3: {color3}</p>
-            <p className="color-minibox">color4: {color4}</p>
-            <p className="color-minibox">color5: {color5}</p>
+            <p className="palette-name" id={this.createPaletteId(i)} >{name}</p>
+            <div className="color-minibox" id={this.createPaletteId(i)} style={{backgroundColor: `#${color1}`}}> </div>
```
Comment (hlhartley) [inline src/components/ProjectCard/ProjectCard.js:26]: I like how you made the key dynamic by using the optional index param in map

File: src/components/ProjectCard/ProjectCard.js [hunk 5]
```diff
@@ -16,31 +18,64 @@ export class ProjectCard extends Component {
...
+    let nameId;
+    if (matchingPalettes.length) {
+      let firstPalette = matchingPalettes[i];
+      nameId = firstPalette.name
+    } else {
+      nameId = null
+    }
+    return nameId
+  }
+
```
Comment (hlhartley) [inline src/components/ProjectCard/ProjectCard.js:47]: Can we go over these together today too? Thanks!

File: src/components/ProjectCard/ProjectCard.scss [hunk 6]
```diff
@@ -2,6 +2,12 @@
...
-    font-size: 20px;
+.color-minibox {
+    border: 1px solid black;
+    height: 20px;
+    width: 20px;
+}
+
+.mapped-palettes {
+    display: flex;
 }
```
Comment (hlhartley) [inline src/components/ProjectCard/ProjectCard.scss:13]: I added some styling for these too. There might be merge conflicts

File: src/containers/App/App.js [hunk 7]
```diff
@@ -6,12 +6,6 @@ import Header from '../../components/Header/Header';
 import ParentSection from '../../components/ParentSection/ParentSection';
 
 export class App extends Component {
-  constructor() {
```
Comment (hlhartley) [inline src/containers/App/App.js:9]: Do we need the constructor in order to access lifecycle methods?

Comment (shannonmoranetz) [reply src/containers/App/App.js:9]: We should have access to that via { Component } !

File: src/components/ProjectCard/ProjectCard.js (hunk 4 above)
Comment (shannonmoranetz) [reply src/components/ProjectCard/ProjectCard.js:26]: Thank you!!! 💅

File: src/components/PaletteControls/PaletteControls.js (hunk 1 above)
Comment (shannonmoranetz) [reply src/components/PaletteControls/PaletteControls.js:14]: Yes absolutely!

File: src/components/PaletteControls/PaletteControls.js (hunk 2 above)
Comment (shannonmoranetz) [reply src/components/PaletteControls/PaletteControls.js:41]: I love that idea!! 👍

**Review by hlhartley — APPROVED** [review]
Reviewed in person

**Pull request merged**